storm-server/src/main/java/org/apache/storm/daemon/nimbus/Nimbus.java

```diff
@@ -2151,9 +2161,10 @@ private void setupBlobstore() throws AuthorizationException, KeyNotFoundExceptio
             store.deleteBlob(toDelete, NIMBUS_SUBJECT);
         }
         LOG.debug("Creating list of key entries for blobstore inside zookeeper {} local {}", activeKeys, activeLocalKeys);
+        CuratorFramework zkClient = BlobStoreUtils.createZKClient(conf);
```

Comment:
**`zkClient` must be guaranteed to be closed.** Please ensure `zkClient` is closed after `for` statement.